internal/services/controllers/v1/task/controller.go

```diff
@@ -819,7 +819,60 @@ func (tc *TasksControllerImpl) handleProcessUserEventTrigger(ctx context.Context
 		})
 	}
 
-	tasks, dags, err := tc.repov1.Triggers().TriggerFromEvents(ctx, tenantId, opts)
+	result, err := tc.repov1.Triggers().TriggerFromEvents(ctx, tenantId, opts)
+
+	if err != nil {
+		return fmt.Errorf("could not trigger tasks from events: %w", err)
+	}
+
+	eventTriggerOpts := make([]tasktypes.CreatedEventTriggerPayloadSingleton, 0)
+	eventSeenAt := time.Now()
+
+	for _, event := range result.Events {
+		for _, task := range result.Tasks {
+			if task.ExternalID.String() == event.TaskExternalId {
+				eventTriggerOpts = append(eventTriggerOpts, tasktypes.CreatedEventTriggerPayloadSingleton{
+					RunId:         task.ID,
+					RunInsertedAt: task.InsertedAt.Time,
+					// FIXME: Should `SeenAt` be set on the SDK when the event is created?
+					EventSeenAt:             eventSeenAt,
+					EventKey:                event.EventKey,
+					EventId:                 event.EventId,
+					EventPayload:            event.EventPayload,
+					EventAdditionalMetadata: event.EventAdditionalMetadata,
+				})
```

Comment:
Probably missing a `break` statement here if we get a match, should we decide to keep this logic

Reply:
I think we need to complete the loop here but I might be misunderstanding, since I was thinking an event can trigger many different tasks or dags, so it wouldn't be safe to break out early b/c we'd lose a triggered run that way